modules/exoplanets/scripts/systemcreation.lua

```diff
@@ -0,0 +1,643 @@
+    Description = string.format(
+      "Terrestrial planets (R < %.2f Earth radii)",
+      MaxRadius.Terrestrial
+    ),
+    Color = { 0.32, 0.2, 0.1 },
+    ColorName = "Brown"
+  },
+  SuperEarth = {
+    Description = string.format(
+      "Super-Earths (%.0f < R < %.0f Earth radii)",
+      MaxRadius.Terrestrial, MaxRadius.SuperEarth
+    ),
+    Color = { 1.0, 0.76, 0.65 },
+    ColorName = "Beige"
+  },
+  NeptuneLike = {
+    Description = string.format(
+      "Neptune-like planets (%.0f < R < %.0f Earth radii)",
+      MaxRadius.SuperEarth, MaxRadius.NeptuneLike
+    ),
+    Color = { 0.22, 0.49, 0.50 },
+    ColorName = "Blue"
+  },
+  GasGiant = {
+    Description = string.format(
+      "Gas giants or larger planets (R > %.0f Earth radii)",
+      MaxRadius.NeptuneLike
+    ),
+    Color = { 0.55, 0.34, 0.39 },
+    ColorName = "Wine red"
+  }
+}
+
+local SizeColorLayerDescription = string.format(
+  [[This layer gives a fixed color to the planet surface based on the
+    planet radius. The planets are split into four categories based on
+    their radius (in Earth radii). 1) %s are %s, 2) %s are %s, 3)
+    %s are %s, and 4) %s are %s.]],
+  PlanetType.Terrestrial.Description, PlanetType.Terrestrial.ColorName,
+  PlanetType.SuperEarth.Description, PlanetType.SuperEarth.ColorName,
+  PlanetType.NeptuneLike.Description, PlanetType.NeptuneLike.ColorName,
+  PlanetType.GasGiant.Description, PlanetType.GasGiant.ColorName
+);
+
+function planetTypeKey(radiusInMeter)
+  if radiusInMeter < MaxRadius.Terrestrial * EarthRadius then
+    return "Terrestrial"
+  elseif radiusInMeter < MaxRadius.SuperEarth * EarthRadius then
+    return "SuperEarth"
+  elseif radiusInMeter < MaxRadius.NeptuneLike * EarthRadius then
+    return "NeptuneLike"
+  else
+    return "GasGiant"
+  end
+end
+
+function hasValue(v)
+  return v ~= nil
+end
+
+-----------------------------------------------------------------------------------
+-- This is the function that adds the scene graph nodes for each exoplanet system.
+-- Edit this to change the visuals of the created exoplanet systems
+-----------------------------------------------------------------------------------
+function addExoplanetSystem(data)
+  if openspace.isEmpty(data) then
+    -- No data was found
+    return
+  end
+
+  local starIdentifier = data.SystemId
+  local guiPath = ExoplanetsGuiPath .. data.StarName
+
+  if openspace.hasSceneGraphNode(starIdentifier) then
+    openspace.printError(
+      "Adding of exoplanet system '" .. data.StarName .. "' failed. " ..
+      "The system has already been added"
+    )
+    return
+  end
+
+  --------------------------------------------------------------------
+  -- Star Globe
+  --------------------------------------------------------------------
+  local starTexture = openspace.propertyValue("Modules.Exoplanets.StarTexture")
+  local starNoDataTexture = openspace.propertyValue("Modules.Exoplanets.NoDataTexture")
+
+  local colorLayers = {}
```

Comment:
I thought of that and made sure to test it as well. But actually, we don't need to.  The Renderable (or in this case, the layer, I think) handles that checking and already provides that check. 

Also, I find that when using the `openspace.printError` function, it's pretty difficult to understand where the output is coming from as it has no "category".  And the resulting error is "uglier", as it adds quotes around the message that is passed to the function. See image below

Instead, in the future, it would be nice to add a check for that on input - to prevent setting properties to invalid files or textures. So that, in this case, you would get an error immediately when starting OpenSpace, as the Module's property value is set. That way, you would also connect it to that property in the Module, and not the texture path that is set for this specific layer

Here's how the error looks if I enter an invalid texture file (which I think is good enough) and how a printed error from `openspace.printError` looks like
![image](https://github.com/user-attachments/assets/3ec1e989-0c45-43b3-a76e-7b35cca5d011)